BoxContentSDK/BoxContentSDK/Views/BOXUserAvatarImageView.m

```diff
@@ -0,0 +1,267 @@
+                                   [UIColor colorWithRed:126.0/255.0f green:213.0/255.0f blue:74.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:175.0/255.0f green:33.0/255.0f blue:244.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:255.0/255.0f green:158.0/255.0f blue:87.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:229.0/255.0f green:67.0/255.0f blue:67.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:93.0/255.0f green:200.0/255.0f blue:167.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:242.0/255.0f green:113.0/255.0f blue:164.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:46.0/255.0f green:113.0/255.0f blue:182.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:226.0/255.0f green:111.0/255.0f blue:60.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:118.0/255.0f green:143.0/255.0f blue:186.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:86.0/255.0f green:193.0/255.0f blue:86.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:239.0/255.0f green:207.0/255.0f blue:46.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:77.0/255.0f green:198.0/255.0f blue:252.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:80.0/255.0f green:23.0/255.0f blue:133.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:238.0/255.0f green:104.0/255.0f blue:50.0/255.0f alpha:1.0f],
+                                   [UIColor colorWithRed:255.0/255.0f green:177.0/255.0f blue:29.0/255.0f alpha:1.0f]];
+    });
+    
+    return avatarBackgroundColors;
+}
+
+- (NSDictionary *)avatarNameAttributes
+{
+    static NSDictionary *avatarNameAttributes = nil;
+    static dispatch_once_t pred;
+    dispatch_once(&pred, ^{
+        NSMutableParagraphStyle *paragraphStyle = [[NSMutableParagraphStyle alloc] init];
+        paragraphStyle.alignment = NSTextAlignmentCenter;
+        avatarNameAttributes = @{ NSForegroundColorAttributeName : [UIColor whiteColor],
+                                  NSFontAttributeName : [UIFont preferredFontForTextStyle:UIFontTextStyleFootnote],
+                                  NSParagraphStyleAttributeName : paragraphStyle };
+    });
+    
+    return avatarNameAttributes;
+}
+
+// This function will take the First word and Last word in the name string as initials.
+// Incorrect initials will be shown when a user has a multi-part last name or a suffix
+// since there is no way of distinguishing whether a user has entered a middle name or falls
+// into the above case. If we instead use initials from the first 2 words, we have the issue of
+// showing incorrect initials when the user has entered a middle name.
+- (NSString *)initialsFromName:(NSString *)name
```

Comment:
This seems like something that must be extracted into a utility method on the user class itself. isn't it?

Reply:
Would normally agree, but we try to keep all the 'BOXModel' classes (such as BOXUser) really clean and free of these kinds of util methods. As much as possible we want those to just represent the API response from the server.